pkg/metrics/history/internal_metrics.go

```diff
@@ -0,0 +1,74 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package history
+
+import (
+	"sync"
+	"time"
+)
+
+// This class collects basic information about the cluster
+// for the web UI's front page.
+// For more detailed metrics collection use Prometheus.
+type InternalMetricsHistory struct {
+	records []*MetricsRecord
+	limit   int
+	mutex   sync.Mutex
+}
+
+type MetricsRecord struct {
+	Timestamp         time.Time
+	TotalApplications int
+	TotalContainers   int
+}
+
+func NewInternalMetricsHistory(limit int) *InternalMetricsHistory {
+	return &InternalMetricsHistory{
+		records: make([]*MetricsRecord, 0),
+		limit:   limit,
+	}
+}
+
+func (h *InternalMetricsHistory) Store(totalApplications, totalContainers int) {
+	h.mutex.Lock()
+	defer h.mutex.Unlock()
+
+	h.records = append(h.records,
```

Comment:
Nice catch. I actually got through that post during my Go tutorial, but didn't realize that I made a mistake here until now.
I started with a containers/list implementation, but got into problems, because it was not typesafe and had to case everywhere where it is used.

Implemented a circular, sized array using the pointer and the limit variables. If there's any official typesafe implementation in Golang, or some popular package to carry out this, please let me know and I'll use that. I think this is the most light implementation that we could use.

Reply:
new impl looks good